game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java

```diff
@@ -118,111 +106,80 @@ private GameParser(
         });
   }
 
-  @Nonnull
-  @VisibleForTesting
-  public static GameData parse(
-      final String mapName,
-      final InputStream stream,
-      final XmlGameElementMapper xmlGameElementMapper)
-      throws GameParseException, EngineVersionException {
-    checkNotNull(mapName);
-    checkNotNull(stream);
-    checkNotNull(xmlGameElementMapper);
-
-    return new GameParser(new GameData(), mapName, xmlGameElementMapper).parse(stream);
-  }
-
   @Nonnull
   private GameData parse(final InputStream stream)
       throws GameParseException, EngineVersionException {
+
+    final Collection<SAXParseException> errorsSax = new ArrayList<>();
     final Element root = XmlReader.parseDom(mapName, stream, errorsSax);
-    parseMapProperties(root);
-    parseMapDetails(root);
-    return data;
-  }
 
-  private GameParseException newGameParseException(final String message) {
-    return newGameParseException(message, null);
-  }
+    if (!errorsSax.isEmpty()) {
+      throw new GameParseException(errorsSax);
+    }
 
-  private GameParseException newGameParseException(
-      final String message, final @Nullable Throwable cause) {
-    final String gameName = data.getGameName() != null ? data.getGameName() : "<unknown>";
-    return new GameParseException(
-        String.format("map name: '%s', game name: '%s', %s", mapName, gameName, message), cause);
+    parseMapPropertiesAndDetails(root);
+    return data;
   }
 
-  private void parseMapProperties(final Element root)
+  private void parseMapPropertiesAndDetails(final Element root)
```

Comment:
Codacy found an issue: [The method 'parseMapPropertiesAndDetails(Element)' has an NPath complexity of 1024, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6147184)